last commit on core/src/va_intel.cpp for  opencv-3.4.3 breaks build on i686 (call of overloaded is ambiguous)

https://github.com/opencv/opencv/commit/4910f16f16a0a0c2b456b14cbc3429c86f96a5f5

##### System information (version)
- OpenCV => 3.4
- Operating System / Platform => Fedora 27 , 28 , 29 and rawhide 
- Compiler => gcc 7.3.1 and gcc 8.2.1


##### Detailed description
opencv-3.4.3 fails to build on i686 see above link 
https://copr.fedorainfracloud.org/coprs/sergiomb/opencv/build/803024/


##### Steps to reproduce

reverting the mention patch , don't fail to build 
 ```
/builddir/build/BUILD/opencv-3.4.3/modules/core/include/opencv2/core/check.hpp:95:85: error: call of overloaded 'check_failed_auto(const uint32_t&, int, const cv::detail::CheckContext&)' is ambiguous
[error.txt](https://github.com/opencv/opencv/files/2421967/error.txt)

BUILDSTDERR:          cv::detail::check_failed_ ## type((v1), (v2), CV__CHECK_LOCATION_VARNAME(id));
```

--- a/modules/core/src/va_intel.cpp
+++ b/modules/core/src/va_intel.cpp
@@ -340,8 +340,8 @@ static void copy_convert_yv12_to_bgr(const VAImage& image, const unsigned char* 
             1.5959997177f
         };
 
-    CV_CheckEQ(image.format.fourcc, VA_FOURCC_YV12, "Unexpected image format");
-    CV_CheckEQ(image.num_planes, 3, "");
+    CV_CheckEQ((size_t)image.format.fourcc, (size_t)VA_FOURCC_YV12, "Unexpected image format");
+    CV_CheckEQ((size_t)image.num_planes, (size_t)3, "");
 
     const size_t srcOffsetY = image.offsets[0];
     const size_t srcOffsetV = image.offsets[1];
@@ -417,8 +417,8 @@ static void copy_convert_bgr_to_yv12(const VAImage& image, const Mat& bgr, unsig
             -0.2909994125f, 0.438999176f, -0.3679990768f, -0.0709991455f
         };
 
-    CV_CheckEQ(image.format.fourcc, VA_FOURCC_YV12, "Unexpected image format");
-    CV_CheckEQ(image.num_planes, 3, "");
+    CV_CheckEQ((size_t)image.format.fourcc, (size_t)VA_FOURCC_YV12, "Unexpected image format");
+    CV_CheckEQ((size_t)image.num_planes, (size_t)3, "");
 
     const size_t dstOffsetY = image.offsets[0];
     const size_t dstOffsetV = image.offsets[1];

